packages/sync-actions/src/products.js

```diff
@@ -58,10 +62,32 @@ function createProductMapActions (mapActionGroup) {
   }
 }
 
+function moveMasterVariantsIntoVariants (before, now) {
+  const move = obj => ({
+    ...obj,
+    masterVariant: undefined,
+    variants: [
+      obj.masterVariant,
+      ...obj.variants || [],
+    ],
+  })
+  const hasMasterVariant = obj => obj && obj.masterVariant
+
+  return [
+    hasMasterVariant(before) ? move(before) : before,
+    hasMasterVariant(now) ? move(now) : now,
```

Comment:
I am not sure how often we will have this scenario but what if the `masterVariant` object is not passed in the `now` object.

Reply:
This is what can not happen in theory. Whenever a project has been created implicitly a master variant will be added. Theoretically we could remove these checks but our test data is sometimes a bit messy. E.g. not having a masterVariant in `now` of `before`.

Reply:
I mean from the user end, I think what you describing is the `before`(what already exists in the CTP)

If we understand each other though 🤔

Reply:
Hm, so I think there are a couple of invariants in this discussions

1. The `before` can be assumed to always have a master variant
2. The `now` should also always have a master variant
   2.1 If it doesn't we can assume that the first variant ought to be the new master variant

If non the the cases hold be should either throw or do nothing which breaks things?